File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/SalaryGradeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SalaryGradeRow } from "./SalaryGradeRow";

export interface SalaryGradeColumns {
    GradeId: Column<SalaryGradeRow>;
    GradeName: Column<SalaryGradeRow>;
    GradeCode: Column<SalaryGradeRow>;
    MinSalary: Column<SalaryGradeRow>;
    MaxSalary: Column<SalaryGradeRow>;
    IsActive: Column<SalaryGradeRow>;
}

export class SalaryGradeColumns extends ColumnsBase<SalaryGradeRow> {
    static readonly columnsKey = 'Operations.SalaryGrade';
    static readonly Fields = fieldsProxy<SalaryGradeColumns>();
}

File: HRMS/HRMS.Web/Modules/Operations/SalaryGrade/SalaryGradeColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.SalaryGrade")]
[BasedOnRow(typeof(SalaryGradeRow), CheckNames = true)]
public class SalaryGradeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int GradeId { get; set; }
    [EditLink, Width(200)]
    public string GradeName { get; set; }
    [Width(100)]
    public string GradeCode { get; set; }
    [Width(120), DisplayFormat("#,##0.00"), AlignRight]
    public decimal MinSalary { get; set; }
    [Width(120), DisplayFormat("#,##0.00"), AlignRight]
    public decimal MaxSalary { get; set; }
    [Width(80)]
    public bool IsActive { get; set; }
}
